### PR TITLE
Feature/e commerce protos

### PR DESCRIPTION
**Pull request by nicopixeldev**
nicopixel - Feature/e commerce protos

### CODE REVIEW

File: staff/nicolas-martin/e-commerce-protos/scripts/cart.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+function Cart() {
```
Comment (maryammdot) [inline staff/nicolas-martin/e-commerce-protos/scripts/cart.js:1]: Cool implementation with the reduce method!!! congrats!

File: staff/nicolas-martin/e-commerce-protos/scripts/cart.test.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
+function createSetOfProducts() {
```
Comment (maryammdot) [inline staff/nicolas-martin/e-commerce-protos/scripts/cart.test.js:1]: cool approximation to a beforeEach!

File: staff/nicolas-martin/e-commerce-protos/scripts/cart.test.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+      
+        expect(cart.products.length === 1, "products should be 1 length");
+        expect(
+          cart.products[0] === tshirt,
+          "expected product " + tshirt + " was not added to cart products"
+        );
+      });
+    
+    // use case 2
+    it("should get the total price of the cart", function() {
```
Comment (maryammdot) [inline staff/nicolas-martin/e-commerce-protos/scripts/cart.test.js]: Be careful with the indentation

File: staff/nicolas-martin/e-commerce-protos/scripts/cart.test.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  cart.add(mobile);
+  var mobile2 = new Mobile(12.22, "Xiaomi", "5X", "space-rose");
+  cart.add(mobile2);
+
+  return cart;
+}
+
+suite("cart", function(){
+
+    describe('*** manage products from Cart ***', function(){
```
Comment (maryammdot) [inline staff/nicolas-martin/e-commerce-protos/scripts/cart.test.js:22]: I would make one describe for each method of the cart and inside each describe I would implement all tests of that method

File: staff/nicolas-martin/e-commerce-protos/scripts/cart.js (hunk 1 above)
Comment (nicopixeldev) [reply staff/nicolas-martin/e-commerce-protos/scripts/cart.js:1]: thankssss! 🍡

File: staff/nicolas-martin/e-commerce-protos/scripts/cart.test.js (hunk 3 above)
Comment (nicopixeldev) [reply staff/nicolas-martin/e-commerce-protos/scripts/cart.test.js]: fixed

File: staff/nicolas-martin/e-commerce-protos/scripts/cart.test.js (hunk 4 above)
Comment (nicopixeldev) [reply staff/nicolas-martin/e-commerce-protos/scripts/cart.test.js:22]: done

**Pull request merged**